File: JavaScript/6js 2/travel.js
/*
|--------------------------------------------------------------------------
| Salesman's Travel
|--------------------------------------------------------------------------
|
*/

function travel(r, zipcode) {
   let x = r.split(",").filter((i) => i.slice(i.indexOf(zipcode), i.length) === zipcode);
   let adr = x.map((i) => i.slice(i.indexOf(" ") + 1, i.length - 9)).join(",");
   let num = x.map((i) => i.slice(0, i.indexOf(" "))).join(",");
   return `${zipcode}:${adr}/${num}`;
}


function travel(r, zipcode) {
   re = RegExp("(\\d+)\\s+(.+)\\s+" + zipcode + "$");
   streets = r
      .split(",")
      .map((x) => x.match(re))
      .filter((x) => x);
   return zipcode + ":" + streets.map((x) => x[2]) + "/" + streets.map((x) => x[1]);
}


function travel(r, zipcode) {
   let matches = r.split(",").filter((str) => str.slice(-8) == zipcode);
   return (
      zipcode +
      ":" +
      matches.map((addr) => addr.split(" ").slice(1, -2).join(" ")).join(",") +
      "/" +
      matches.map((addr) => addr.split(" ")[0]).join(",")
   );
}

File: Python/6py 2/travel.py
"""
|--------------------------------------------------------------------------
| Salesman's Travel
|--------------------------------------------------------------------------
|
"""

import re
def travel(r, zipcode):
   x = re.findall(r'(\d+) (.+?) ([A-Z]{2} \d+)', r)
   return f'{zipcode}:{",".join(i[1] for i in x if i[2]==zipcode)}/{",".join(i[0] for i in x if i[2]==zipcode)}'


def travel(r, zipcode):
   streets = []
   nums = []
   addresses = r.split(',')
   for address in addresses:
      if ' '.join(address.split()[-2:]) == zipcode:
         streets.append(' '.join(address.split()[1:-2]))
         nums += address.split()[:1]
   return '{}:{}/{}'.format(zipcode, ','.join(streets), ','.join(nums))


import re
def travel(r, zipcode):
   res = [(m.group(2), m.group(1)) for m in re.finditer(r'(\d+) ([^,]+) ([A-Z][A-Z] \d{5})', r) if zipcode == m.group(3)]
   return '{}:{}/{}'.format(zipcode, ','.join(a[0] for a in res), ','.join(a[1] for a in res))